Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
-							BLUE_LOG(warning) << "#" << swept_surface->getId() << " IfcSweptSurface without placement found.";
-						}
+					EXPRESSReference<typename IfcEntityTypesT::IfcCurve>& outerBoundary = surface->OuterBoundary;
 
-						carve::math::Matrix swept_surface_matrix(pos);
-						if(swept_surface_placement) {
-							swept_surface_matrix = pos * placementConverter->convertIfcAxis2Placement3D(swept_surface_placement);
-						}
+					if(outerBoundary) {
+					}
 
-						// (1/2) IfcSurfaceOfLinearExtrusion SUBTYPE of IfcSweptSurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcSurfaceOfLinearExtrusion> linear_extrusion =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcSurfaceOfLinearExtrusion>(swept_surface);
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCurve>> vInnerBoundaries;
+					vInnerBoundaries.resize(surface->InnerBoundaries.size());
 
-						if(linear_extrusion) {
-							// Get extrude direction and depth.
-							std::shared_ptr<typename IfcEntityTypesT::IfcDirection>& linear_extrusion_direction = linear_extrusion->ExtrudedDirection.lock();
-							double linear_extrusion_depth = linear_extrusion->Depth;
+					std::transform(
+						surface->InnerBoundaries.begin(),
+						surface->InnerBoundaries.end(),
+						vInnerBoundaries.begin(),
+						[](auto& it) { return it.lock(); }
+						);
 
-							// TODO: implement		// Interpret values and calculate.
-#ifdef _DEBUG
-							std::cout << "Warning\t| IfcSurfaceOfLinearExtrusion not implemented." << std::endl;
-#endif
-							return;
-						}
+					for(unsigned int i = 0; i < vInnerBoundaries.size(); ++i) {
+						EXPRESSReference<typename IfcEntityTypesT::IfcCurve>& innerCurve = vInnerBoundaries[i];
+					}
 
-						// (2/2) IfcSurfaceOfRevolution SUBTYPE of IfcSweptSurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcSurfaceOfRevolution> surface_of_revolution =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcSurfaceOfRevolution>(swept_surface);
+					*/
 
-						if(surface_of_revolution) {
-							// Get axis position.
-							std::shared_ptr<typename IfcEntityTypesT::IfcAxis1Placement>& axis_position =
-								surface_of_revolution->AxisPosition.lock(); // TODO: check member name erledigt
+					throw oip::UnhandledException(surface);
+				}
 
-	  // TODO: implement		// Interpret values and calculate.
-#ifdef _DEBUG
-							std::cout << "Warning\t| IfcSurfaceOfRevolution not implemented." << std::endl;
-#endif
-							return;
+				/*! \brief  Converts \c IfcCurveBoundedSurface to ...
+				\param	surface	\c IfcCurveBoundedSurface entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcCurveBoundedSurface is a subtype of \c IfcBoundedSurface.
+				*/
+
+				void convertIfcCurveBoundedSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcCurveBoundedSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {

```

Comment:
Same, please make sure that pos is returned as out parameter. And maybe also polyline data?

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1